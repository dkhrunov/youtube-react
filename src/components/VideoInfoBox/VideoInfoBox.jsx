import React, { useState } from 'react';
import { Image, Button } from 'semantic-ui-react';
import './VideoInfoBox.scss';

const VideoInfoBox = (props) => {
	const [collapsed, setCollapsed] = useState(true);

	const onToggleCollapseButtonClick = () => {
		setCollapsed((prevCollapsed) => !prevCollapsed);
	};

	let descriptionTextClass = collapsed ? 'collapsed' : 'expanded';
	let buttonTitle = collapsed ? 'Show More' : 'Show Less';

	return (
		<div className='video-info-box'>
			<Image
				className='channek-image'
				src='http://via.placeholder.com/48x48'
				circular
			/>
			<div className='video-info'>
				<div className='channel-name'>Channel Name</div>
				<div className='video-publication-date'>Thu 24, 2018</div>
			</div>
			<Button color='youtube'>91.5k Subscribe</Button>
			<div className='video-description'>
				<div className={descriptionTextClass}>
					<p>Paragraph 1</p>
					<p>Paragraph 2</p>
					<p>Paragraph 3</p>
					<p>Paragraph 4</p>
					<p>Paragraph 5</p>
				</div>
				<Button compact onClick={onToggleCollapseButtonClick}>
					{buttonTitle}
				</Button>
			</div>
		</div>
	);
};

export default VideoInfoBox;
