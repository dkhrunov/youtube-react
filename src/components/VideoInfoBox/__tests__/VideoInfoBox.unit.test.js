import React from 'react';
import { shallow, mount } from 'enzyme';
import VideoInfoBox from '../VideoInfoBox';

describe('Renders VideoInfoBox component', () => {
	test('renders collapsed comments block', () => {
		const wrapper = shallow(<VideoInfoBox />);
		expect(wrapper).toMatchSnapshot();
	});

	test('renders expanded comments block', () => {
		const wrapper = mount(<VideoInfoBox />);
		const button = wrapper.find('.video-description button');
		button.simulate('click');
		expect(wrapper).toMatchSnapshot();
	});
});
