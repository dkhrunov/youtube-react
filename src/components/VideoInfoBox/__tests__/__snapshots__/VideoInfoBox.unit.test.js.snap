// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders VideoInfoBox component renders collapsed 1`] = `
<div
  className="video-info-box"
>
  <Image
    as="img"
    circular={true}
    className="channek-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Channel Name
    </div>
    <div
      className="video-publication-date"
    >
      Thu 24, 2018
    </div>
  </div>
  <Button
    as="button"
    color="youtube"
  >
    91.5k Subscribe
  </Button>
  <div
    className="video-description"
  >
    <div
      className="collapsed"
    >
      <p>
        Paragraph 1
      </p>
      <p>
        Paragraph 2
      </p>
      <p>
        Paragraph 3
      </p>
      <p>
        Paragraph 4
      </p>
      <p>
        Paragraph 5
      </p>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show More
    </Button>
  </div>
</div>
`;

exports[`Renders VideoInfoBox component renders collapsed comments block 1`] = `
<div
  className="video-info-box"
>
  <Image
    as="img"
    circular={true}
    className="channek-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Channel Name
    </div>
    <div
      className="video-publication-date"
    >
      Thu 24, 2018
    </div>
  </div>
  <Button
    as="button"
    color="youtube"
  >
    91.5k Subscribe
  </Button>
  <div
    className="video-description"
  >
    <div
      className="collapsed"
    >
      <p>
        Paragraph 1
      </p>
      <p>
        Paragraph 2
      </p>
      <p>
        Paragraph 3
      </p>
      <p>
        Paragraph 4
      </p>
      <p>
        Paragraph 5
      </p>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show More
    </Button>
  </div>
</div>
`;

exports[`Renders VideoInfoBox component renders expanded 1`] = `
<div
  className="video-info-box"
>
  <img
    className="ui circular image channek-image"
    src="http://via.placeholder.com/48x48"
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Channel Name
    </div>
    <div
      className="video-publication-date"
    >
      Thu 24, 2018
    </div>
  </div>
  <button
    className="ui youtube button"
    onClick={[Function]}
  >
    91.5k Subscribe
  </button>
  <div
    className="video-description"
  >
    <div
      className="expanded"
    >
      <p>
        Paragraph 1
      </p>
      <p>
        Paragraph 2
      </p>
      <p>
        Paragraph 3
      </p>
      <p>
        Paragraph 4
      </p>
      <p>
        Paragraph 5
      </p>
    </div>
    <button
      className="ui compact button"
      onClick={[Function]}
    >
      Show Less
    </button>
  </div>
</div>
`;

exports[`Renders VideoInfoBox component renders expanded comments block 1`] = `
<div
  className="video-info-box"
>
  <img
    className="ui circular image channek-image"
    src="http://via.placeholder.com/48x48"
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Channel Name
    </div>
    <div
      className="video-publication-date"
    >
      Thu 24, 2018
    </div>
  </div>
  <button
    className="ui youtube button"
    onClick={[Function]}
  >
    91.5k Subscribe
  </button>
  <div
    className="video-description"
  >
    <div
      className="expanded"
    >
      <p>
        Paragraph 1
      </p>
      <p>
        Paragraph 2
      </p>
      <p>
        Paragraph 3
      </p>
      <p>
        Paragraph 4
      </p>
      <p>
        Paragraph 5
      </p>
    </div>
    <button
      className="ui compact button"
      onClick={[Function]}
    >
      Show Less
    </button>
  </div>
</div>
`;

exports[`Renders VideoInfoBox component renders: collapsed comments block 1`] = `
<div
  className="video-info-box"
>
  <Image
    as="img"
    circular={true}
    className="channek-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Channel Name
    </div>
    <div
      className="video-publication-date"
    >
      Thu 24, 2018
    </div>
  </div>
  <Button
    as="button"
    color="youtube"
  >
    91.5k Subscribe
  </Button>
  <div
    className="video-description"
  >
    <div
      className="collapsed"
    >
      <p>
        Paragraph 1
      </p>
      <p>
        Paragraph 2
      </p>
      <p>
        Paragraph 3
      </p>
      <p>
        Paragraph 4
      </p>
      <p>
        Paragraph 5
      </p>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show More
    </Button>
  </div>
</div>
`;
